wasabi/printer.py

```diff
@@ -92,14 +93,36 @@ def good(
             title, text, style=MESSAGES.GOOD, show=show, spaced=spaced, exits=exits
         )
 
+    @overload
     def fail(
         self,
         title: Any = "",
         text: Any = "",
         show: bool = True,
         spaced: bool = False,
-        exits: Optional[int] = None,
+        exits: Optional[Literal[0, False]] = None,
     ):
+        ...
+
+    @overload
+    def fail(
+        self,
+        title: Any = "",
+        text: Any = "",
+        show: bool = True,
+        spaced: bool = False,
+        exits: Literal[1, True] = True,
+    ) -> NoReturn:
+        ...
```

Comment:
I think all the defaults here can be `...` instead?

Reply:
Unresolving these and putting these back to regular defaults. I know this is not super helpful in Mypy but if you're using VSCode autocomplete (and I assume other pyright editors) you see the `...` instead of the actual defaults here.

Reply:
Does pyright detect if the defaults get out of sync?

I tested with mypy and I could make all the defaults different with no issues found by mypy, while only the defaults from the main method are used in practice.

My guess is that vscode would happily show you completely incorrect defaults in this case?

I don't really mean to hold up this particular (relatively minor) change, it's just very frustrating in general.